@xen-orchestra/lite/src/components/Table.tsx

```diff
@@ -19,55 +52,318 @@ interface ParentState {}
 
 interface State {}
 
-interface Props {
-  collection: Item[] | undefined
-  columns: Column<any>[]
-  placeholder?: JSX.Element
+interface StatePagination {}
+
+type Props = IsSelectable &
+  RouteComponentProps & {
+    collection: Item[] | undefined
+    columns: Column<any>[]
+    dataType?: string
+    stateUrlParam: string
+  }
+
+interface PropsPagination {
+  dataType?: string
+  nItemsOnThePage: number
+  nSelectedItems?: number
+  nTotalItems: number
+  nTotalPage: number
+  onPaginationChange: (_: any, page: number) => void
+  onShowByChange: (value: SelectChangeEvent<unknown>) => void
+  page: number
+  reverse?: boolean
+  showByValue: number
 }
 
 interface ParentEffects {}
 
-interface Effects {}
+interface Effects {
+  handlePaginationChange: (e: any, page: number) => void
+  handleRowPerPage: (e: any) => void
+  toggleAllSelectedItem: () => void
+  toggleSelectItem: (item: any) => void
+}
+interface EffectsPagination {}
+
+interface Computed {
+  page: number
+  paginatedCollection: Props['collection']
+  rowsPerPage: number
+}
 
-interface Computed {}
+interface ComputedPagination {}
 
-const StyledTable = styled.table`
-  border: 1px solid #333;
-  td {
-    border: 1px solid #333;
-  }
-  thead {
-    background-color: #333;
-    color: #fff;
+const StyledSelect = styled(Select)(({ theme }) => ({
+  backgroundColor: theme.palette.lightGray.main,
```

Comment:
```suggestion
  backgroundColor: palette['gray'].[palette.mode]
```
The light/dark mode must be taken into account.